airflow-core/src/airflow/migrations/versions/0071_3_1_0_make_bundle_name_not_nullable.py

```diff
@@ -0,0 +1,94 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Make bundle_name not nullable.
+
+Revision ID: 1b612ec87098
+Revises: 0242ac120002
+Create Date: 2025-05-20 21:13:32.525404
+
+"""
+
+from __future__ import annotations
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.sql import text
+
+# revision identifiers, used by Alembic.
+revision = "1b612ec87098"
+down_revision = "0242ac120002"
+branch_labels = None
+depends_on = None
+airflow_version = "3.1.0"
+
+
+def upgrade():
+    """Apply Make bundle_name not nullable."""
+    dialect_name = op.get_bind().dialect.name
+    if dialect_name == "postgresql":
```

Comment:
This won’t work in offline mode.

Reply:
Could you plz elaborate this? I think it's used in many other migrations, like [here](https://github.com/gyli/airflow/blob/main/airflow-core/src/airflow/migrations/versions/0060_3_0_0_add_try_id_to_ti_and_tih.py#L101).

I've also tested with offline mode with command `alembic -c airflow-core/src/airflow/alembic.ini upgrade heads --sql` in different backend like `breeze --backend postgres`, and looks like the returned sql follows the correct dialect.

Reply:
@uranusjr , I think it will work since there's no DB access and only an If statement